src/io/jesi/backpack/collection.cljc

```diff
@@ -371,3 +371,19 @@
   [m]
   (= (count m)
      (count (set (vals m)))))
+
+(defn filter-by
+  "Filters a collection where a key matches a predicate
+  e.g.
+  (let [m [{:id 1} {:id 2}]
+    (filter-by m :id (bp/p= 1))) ; returns {:id 1}"
+  [key-fn pred coll]
+  (filter (comp pred key-fn) coll))
+
+(defn filter-by=
```

Comment:
```suggestion
(defn filter-key=
```

Would `filter-key=` be a better name?

Reply:
yeah, was struggling with naming this one. done